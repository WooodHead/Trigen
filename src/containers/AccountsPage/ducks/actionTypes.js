export const ADD_ACCOUNT = 'ADD_ACCOUNT';
export const FETCH_ACCOUNTS = 'FETCH_ACCOUNTS';
export const FETCH_ACCOUNTS_SUCCESS = 'FETCH_ACCOUNTS_SUCCESS';
export const FETCH_ACCOUNTS_ERROR = 'FETCH_ACCOUNTS_ERROR';
export const EDIT_ACCOUNT = 'EDIT_ACCOUNT';
export const UPDATE_ACCOUNT = 'UPDATE_ACCOUNT';
export const UPDATE_ACCOUNT_SUCCESS = 'UPDATE_ACCOUNT_SUCCESS';
export const UPDATE_ACCOUNT_ERROR = 'UPDATE_ACCOUNT_ERROR';
export const SAVE_NEW_ACCOUNT = 'SAVE_NEW_ACCOUNT';
export const SAVE_NEW_ACCOUNT_SUCCESS = 'SAVE_NEW_ACCOUNT_SUCCESS';
export const SAVE_NEW_ACCOUNT_ERROR = 'SAVE_NEW_ACCOUNT_ERROR';
export const CANCEL_EDIT = 'CANCEL_EDIT';
export const CHANGE_ACCOUNT = 'CHANGE_ACCOUNT';
export const DELETE_ACCOUNT = 'DELETE_ACCOUNT';
export const DELETE_ACCOUNT_SUCCESS = 'DELETE_ACCOUNT_SUCCESS';
export const DELETE_ACCOUNT_ERROR = 'DELETE_ACCOUNT_ERROR';
export const FILTER_ACCOUNT = 'FILTER_ACCOUNT';
