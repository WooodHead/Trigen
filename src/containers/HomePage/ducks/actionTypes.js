export const GENERATE_PASSWORD = 'GENERATE_PASSWORD';
export const GENERATE_PASSWORD_SUCCESS = 'GENERATE_PASSWORD_SUCCESS';
export const GENERATE_PASSWORD_ERROR = 'GENERATE_PASSWORD_ERROR';
export const CLEAR_PASSWORD = 'CLEAR_PASSWORD';
