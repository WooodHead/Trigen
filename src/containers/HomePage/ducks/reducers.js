import { combineReducers } from 'redux';


export function PasswordGenReducer(state = {}, action) {
  switch(action.type) {
    default:
      return state;
  }
}
