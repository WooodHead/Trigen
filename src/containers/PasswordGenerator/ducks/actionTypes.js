export const GENERATE_PASSWORD = 'GENERATE_PASSWORD';
export const GENERATE_PASSWORD_SUCCESS = 'GENERATE_PASSWORD_SUCCESS';
export const GENERATE_PASSWORD_ERROR = 'GENERATE_PASSWORD_ERROR';
export const CLEAR_PASSWORD = 'CLEAR_PASSWORD';
export const START_PROGRESS = 'START_PROGRESS';
export const STOP_PROGRESS = 'STOP_PROGRESS';
