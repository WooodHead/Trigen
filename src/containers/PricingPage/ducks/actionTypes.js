export const BUY_SLOTS = 'BUY_SLOTS';
export const BUY_SLOTS_SUCCESS = 'BUY_SLOTS_SUCCESS';
export const BUY_SLOTS_ERROR = 'BUY_SLOTS_ERROR';
