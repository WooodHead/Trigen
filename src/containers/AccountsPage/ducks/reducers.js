import { combineReducers } from 'redux';

import * as actionTypes from './actionTypes';


// Reducer
export function AccountListReducer(state = [], action) {
  switch(action.type) {
    case actionTypes.FETCH_ACCOUNTS_SUCCESS: {
      const { response, genIDFunc } = action.payload;
      const accounts = response.data;
      const accountList = accounts.map(account => {
        return {
          ...account,
          fakeID: genIDFunc(),
          edit: false,
        };
      });

      return Array.concat([], accountList);
    }

    case actionTypes.ADD_ACCOUNT: {
      console.log(action.payload);
      return Array.concat([], [{ ...action.payload.data }], state);
    }

    case actionTypes.EDIT_ACCOUNT: {
      const { fakeID, data } = action.payload;
      const newArr = state.slice(0);
      const index = newArr.findIndex(val => {
        return val.fakeID === fakeID;
      });
      newArr[index] = { ...newArr[index], ...data };

      return newArr;
    }

    case actionTypes.SAVE_NEW_ACCOUNT_SUCCESS:
    case actionTypes.SAVE_ACCOUNT: {
      const account = action.payload;
      const newArr = state.slice(0);
      const index = newArr.findIndex(val => {
        return val.fakeID === account.fakeID;
      });
      newArr[index] = { ...account, new: false, edit: false };

      return newArr;
    }

    case actionTypes.CANCEL_EDIT: {
      const { fakeID } = action.payload;
      let newArr = state.slice(0);
      const index = newArr.findIndex(val => {
        return val.fakeID === fakeID;
      });

      newArr[index] = { ...newArr[index], edit: false };

      if(
        newArr[index].new &&
        newArr[index].username === '' &&
        newArr[index].domain === ''
      ) {
        newArr = [ ...newArr.slice(0, index), ...newArr.slice(index + 1)];
      }

      return newArr;
    }

    case actionTypes.DELETE_ACCOUNT: {
      const { fakeID } = action.payload;
      const index = state.findIndex(val => {
        return val.fakeID === fakeID;
      });

      const newArr = [ ...state.slice(0, index), ...state.slice(index + 1)];
      return newArr;
    }

    default:
      return state;
  }
}

export function AccountStatusReducer(state = {
  fetched: false,
  errored: false,
}, action) {
  switch(action.type) {
    case actionTypes.FETCH_ACCOUNTS_SUCCESS: {
      return { ...state, fetched: true };
    }

    case actionTypes.FETCH_ACCOUNTS_ERROR: {
      return {
        ...state,
        fetched: false,
        errored: true,
      };
    }

    default: {
      return state;
    }
  }
}

export function EditAccountListReducer(state = {}, action) {
  switch(action.type) {
    case actionTypes.ADD_ACCOUNT: {
      const { data } = action.payload;

      return { ...state, [data.fakeID]: data};
    }

    case actionTypes.EDIT_ACCOUNT: {
      const { fakeID, data } = action.payload;

      return { ...state, [fakeID]: data };
    }

    case actionTypes.SAVE_ACCOUNT: {
      const account = action.payload;
      delete state[account.fakeID];

      return { ...state };
    }

    case actionTypes.CANCEL_EDIT: {
      const { fakeID } = action.payload;
      delete state[fakeID];

      return { ...state };
    }

    case actionTypes.CHANGE_ACCOUNT: {
      const { fakeID, target } = action.payload;

      return Object.assign({}, state, {
        [fakeID]: Object.assign({}, state[fakeID], {
          [target.name]: target.value,
        }),
      });
    }

    default:
      return state;
  }
}

export function AccountFilterReducer( state = '', action) {
  switch(action.type) {
    case actionTypes.FILTER_ACCOUNT: {
      const { filter } = action.payload;
      return filter;
    }

    default: {
      return state;
    }
  }
}

const AccountReducer = combineReducers({
  Accounts: AccountListReducer,
  EditedAccounts: EditAccountListReducer,
  AccountFilter: AccountFilterReducer,
  AccountStatus: AccountStatusReducer,
});

export default AccountReducer;
