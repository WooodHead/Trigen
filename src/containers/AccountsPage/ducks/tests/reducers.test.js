import { AccountListReducer, EditAccountListReducer } from '../reducers';
import * as types from '../actionTypes';


describe('Reducers', () => {
  describe('AccountListReducer', () => {
    const INITIAL_STATE = [
      {
        fakeID: 'fakeID',
        id: 0,
        username: 'ttester',
        domain: 'example.com',
        edit: false,
      }
    ];

    it('should return unmodified state as default', () => {
      expect(AccountListReducer(undefined, {}))
        .toEqual([]);
    });

    it('should handle GET_ACCOUNTS', () => {
      const genIDMock = jest.fn(() => 'fakeID');
      expect(AccountListReducer(undefined, {
        type: types.GET_ACCOUNTS,
        payload: {
          accounts: [
            {
              id: 0,
              username: 'ttester',
              domain: 'example.com',
            }
          ],
          genID: genIDMock
        }
      })).toMatchSnapshot();
      expect(genIDMock).toHaveBeenCalled();
    });

    it('should handle ADD_ACCOUNT', () => {
      expect(AccountListReducer([], {
        type: types.ADD_ACCOUNT,
        payload: { data : {
          test: 'test',
        }}
      })).toMatchSnapshot();
    });

    it('should handle EDIT_ACCOUNT', () => {
      const action = {
        type: types.EDIT_ACCOUNT,
        payload: { fakeID: 'fakeID', data: {
          username: 'dudeDudsen',
          domain: 'example.com',
          edit: true,
        }}
      };
      expect(AccountListReducer(INITIAL_STATE, action)).toMatchSnapshot();
    });

    it('should handle SAVE_ACCOUNT', () => {
      const initialState = [ { ...INITIAL_STATE[0], edit: true } ];
      const action = {
        type: types.SAVE_ACCOUNT,
        payload: {
          ...INITIAL_STATE[0],
          username: 'dudeDudsen',
        }
      };

      expect(AccountListReducer(initialState, action)).toMatchSnapshot();
    });

    it('should handle CANCEL_EDIT with existing account', () => {
      const action = {
        type: types.CANCEL_EDIT,
        payload: { fakeID: 'fakeID' },
      };

      expect(AccountListReducer(INITIAL_STATE, action))
        .toMatchSnapshot();
    });

    it('should handle CANCEL_EDIT with empty new account', () => {
      const initialState = [ {
        ...INITIAL_STATE[0],
        new: true,
        domain: '',
        username: '',
      } ];
      const action = {
        type: types.CANCEL_EDIT,
        payload: { fakeID: 'fakeID' }
      };

      expect(AccountListReducer(initialState, action))
        .toMatchSnapshot();
    });

    it('should handle DELETE_ACCOUNT', () => {
      const action = {
        type: types.DELETE_ACCOUNT,
        payload: { fakeID: 'fakeID' }
      };

      expect(AccountListReducer(INITIAL_STATE, action))
        .toMatchSnapshot();
    })
  });

  describe('EditAccountListReducer', () => {
    it('should return unmodified state as default', () => {
      expect(EditAccountListReducer(undefined, {}))
        .toMatchSnapshot();
    });

    it('should handle ADD_ACCOUNT', () => {
      const action = {
        type: types.ADD_ACCOUNT,
        payload: { data: {
          fakeID: 'fakeID',
          test: 'test',
        }}
      };

      expect(EditAccountListReducer({}, action))
        .toMatchSnapshot();
    });

    it('should handle EDIT_ACCOUNT', () => {
      const action = {
        type: types.EDIT_ACCOUNT,
        payload: { fakeID: 'fakeID', data: {
          fakeID: 'fakeID',
          test: 'test',
        }}
      };

      expect(EditAccountListReducer({}, action))
        .toMatchSnapshot();
    });

    it('should handle SAVE_ACCOUNT', () => {
      const initialState = {
        'fakeID': {
          test: 'test'
        }
      };
      const action = {
        type: types.SAVE_ACCOUNT,
        payload: { fakeID: 'fakeID' }
      };

      expect(EditAccountListReducer(initialState, action))
        .toMatchSnapshot();
    });

    it('should handle CANCEL_EDIT', () => {
      const initialState = { 'fakeID': { test: 'test' }};
      const action = {
        type: types.CANCEL_EDIT,
        payload: { fakeID: 'fakeID' }
      };

      expect(EditAccountListReducer(initialState, action))
        .toMatchSnapshot();
    });

    it('should handle CHANGE_ACCOUNT', () => {
      const initialState = {
        'fakeID': {
          fakeID: 'fakeID',
          username: 'ttester',
          domain: 'example.com',
        }
      };
      const action = {
        type: types.CHANGE_ACCOUNT,
        payload: {
          fakeID: 'fakeID',
          target: {
            name: 'username',
            value: 'dudeDudsen',
          }
        }
      };

      expect(EditAccountListReducer(initialState, action))
        .toMatchSnapshot();
    });
  });
});
