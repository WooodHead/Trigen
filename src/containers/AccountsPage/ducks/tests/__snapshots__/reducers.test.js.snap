exports[`Reducers AccountListReducer should handle ADD_ACCOUNT 1`] = `
Array [
  Object {
    "test": "test",
  },
]
`;

exports[`Reducers AccountListReducer should handle CANCEL_EDIT with empty new account 1`] = `Array []`;

exports[`Reducers AccountListReducer should handle CANCEL_EDIT with existing account 1`] = `
Array [
  Object {
    "domain": "example.com",
    "edit": false,
    "fakeID": "fakeID",
    "id": 0,
    "username": "ttester",
  },
]
`;

exports[`Reducers AccountListReducer should handle DELETE_ACCOUNT_SUCCESS 1`] = `
Array [
  Object {
    "domain": "example.com",
    "edit": false,
    "fakeID": "fakeID",
    "id": 0,
    "username": "ttester",
  },
]
`;

exports[`Reducers AccountListReducer should handle EDIT_ACCOUNT 1`] = `
Array [
  Object {
    "domain": "example.com",
    "edit": true,
    "fakeID": "fakeID",
    "id": 0,
    "username": "dudeDudsen",
  },
]
`;

exports[`Reducers AccountListReducer should handle FETCH_ACCOUNTS_SUCCESS 1`] = `
Array [
  Object {
    "domain": "example.com",
    "edit": false,
    "fakeID": "fakeID",
    "id": 0,
    "username": "ttester",
  },
]
`;

exports[`Reducers AccountListReducer should handle SAVE_NEW_ACCOUNT_SUCCESS 1`] = `
Array [
  Object {
    "domain": "example.com",
    "edit": false,
    "fakeID": "fakeID",
    "id": 0,
    "new": false,
    "username": "dudeDudsen",
  },
]
`;

exports[`Reducers AccountListReducer should handle UPDATE_ACCOUNT_SUCCESS 1`] = `
Array [
  Object {
    "domain": "example.com",
    "edit": false,
    "fakeID": "fakeID",
    "id": 0,
    "new": false,
    "username": "dudeDudsen",
  },
]
`;

exports[`Reducers AccountListStatusReducer should handle FETCH_ACCOUNTS 1`] = `
Object {
  "errored": false,
  "fetched": false,
  "isLoading": true,
}
`;

exports[`Reducers AccountListStatusReducer should handle FETCH_ACCOUNTS_ERROR 1`] = `
Object {
  "errored": true,
  "fetched": false,
  "isLoading": false,
}
`;

exports[`Reducers AccountListStatusReducer should handle FETCH_ACCOUNTS_SUCCESS 1`] = `
Object {
  "errored": false,
  "fetched": true,
  "isLoading": false,
}
`;

exports[`Reducers AccountListStatusReducer should handle authTypes.LOGOUT 1`] = `
Object {
  "errored": false,
  "fetched": false,
  "isLoading": false,
}
`;

exports[`Reducers AccountStatusReducer should handle DELETE_ACCOUNT 1`] = `
Object {
  "fake123": Object {
    "isLoading": true,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle DELETE_ACCOUNT_ERROR 1`] = `
Object {
  "fake123": Object {
    "isLoading": false,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle DELETE_ACCOUNT_SUCCESS 1`] = `
Object {
  "fake123": Object {
    "isLoading": false,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle SAVE_NEW_ACCOUNT 1`] = `
Object {
  "fake123": Object {
    "isLoading": true,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle SAVE_NEW_ACCOUNT_ERROR 1`] = `
Object {
  "fake123": Object {
    "isLoading": false,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle SAVE_NEW_ACCOUNT_SUCCESS 1`] = `
Object {
  "fake123": Object {
    "isLoading": false,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle UPDATE_ACCOUNT 1`] = `
Object {
  "fake123": Object {
    "isLoading": true,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle UPDATE_ACCOUNT_ERROR 1`] = `
Object {
  "fake123": Object {
    "isLoading": false,
  },
}
`;

exports[`Reducers AccountStatusReducer should handle UPDATE_ACCOUNT_SUCCESS 1`] = `
Object {
  "fake123": Object {
    "isLoading": false,
  },
}
`;

exports[`Reducers EditAccountListReducer should handle ADD_ACCOUNT 1`] = `
Object {
  "fakeID": Object {
    "fakeID": "fakeID",
    "test": "test",
  },
}
`;

exports[`Reducers EditAccountListReducer should handle CANCEL_EDIT 1`] = `Object {}`;

exports[`Reducers EditAccountListReducer should handle CHANGE_ACCOUNT 1`] = `
Object {
  "fakeID": Object {
    "domain": "example.com",
    "fakeID": "fakeID",
    "username": "dudeDudsen",
  },
}
`;

exports[`Reducers EditAccountListReducer should handle EDIT_ACCOUNT 1`] = `
Object {
  "fakeID": Object {
    "fakeID": "fakeID",
    "test": "test",
  },
}
`;

exports[`Reducers EditAccountListReducer should handle SAVE_ACCOUNT 1`] = `
Object {
  "fakeID": Object {
    "test": "test",
  },
}
`;

exports[`Reducers EditAccountListReducer should return unmodified state as default 1`] = `Object {}`;
